presto-native-execution/presto_cpp/main/TaskResource.h

```diff
@@ -65,7 +65,7 @@ class TaskResource {
       const std::vector<std::string>& pathMatch);
 
   TaskManager& taskManager_;
-  std::unique_ptr<velox::memory::ScopedMemoryPool> pool_;
+  std::shared_ptr<velox::memory::MemoryPool> pool_;
```

Comment:
Why do we have shared instead of unique pointers now?

Reply:
The child memory pool holds a shared reference to its parent. So we need to use shared ptr to maintain the memory pool hierarchy.